tests/test-obsidian.el

```diff
@@ -2,35 +2,43 @@
 (require 'obsidian)
 (require 'buttercup)
 
-(defvar obsidian--test-dir "./tests/test_vault")
+;; Using a relative path for obsidian-change-vault/obsidian-change-vault will
+;; result in different values for obsidian-directory depending on the directory
+;; of the most recently visited file
+(defvar obsidian--test-dir (expand-file-name "./tests/test_vault"))
 (defvar obsidian--test--original-dir (or obsidian-directory obsidian--test-dir))
-(defvar obsidian--test--original-tags-list obsidian--tags-list)
 (defvar obsidian--test-number-of-tags 9)
 (defvar obsidian--test-number-of-visible-tags 6)
 (defvar obsidian--test-number-of-notes 11)
 (defvar obsidian--test-number-of-visible-notes 9)
 (defvar obsidian--test-number-of-visible-directories 2)
+(defvar obsidian--test-visibility-cfg obsidian-include-hidden-files)
+
+(defun obsidian-test--delete-all-test-files ()
+  "Function to delete all files potentially left behind by tests."
+  (delete-file (concat obsidian-directory "/foo.md"))
+  (delete-file (concat obsidian-directory "/bar.md"))
+  (delete-file (concat obsidian-directory "/inbox/foo.md"))
+  (delete-file (concat obsidian-directory "/inbox/bar.md"))
+  (delete-file (concat obsidian-directory "/subdir/foo.md"))
+  (delete-file (concat obsidian-directory "/subdir/bar.md")))
+
+(defun obsidian-test--jump-to-file (file)
+  "FILE is a path relative to the obsidian vault."
+  (let* ((obsidian-inbox-directory "inbox")
+         (executing-kbd-macro t)
+         (unread-command-events (listify-key-sequence (format "%s\n" file))))
+    (call-interactively #'obsidian-jump)))
 
 (describe "check path setting"
-  (before-all (obsidian-specify-path obsidian--test-dir))
-  (after-all (obsidian-specify-path obsidian--test--original-dir))
-
+  (before-all (obsidian-change-vault obsidian--test-dir))
```

Comment:
`obsidian-change-vault` updates the cache as well as changing the vault path, so `obsidian-specify-path` was replaced with that function.